plugins/callback/foreman.py

```diff
@@ -248,12 +323,45 @@ def send_facts(self):
                 },
             }
 
-            self._send_data('facts', host, facts)
+            self._send_data('facts', 'foreman', host, facts)
 
-    def send_reports(self, stats):
+    def send_reports_proxy_host_report(self, stats):
+        """
+        Send reports to Foreman Smart Proxy running Host Reports
+        plugin. The format is native Ansible report without any
+        changes.
+        """
+        for host in stats.processed.keys():
+            total = stats.summarize(host)
+            report = {
+                "host": host,
+                "reported_at": get_now(),
+                "uuid": str(uuid.uuid1()),
```

Comment:
This is just an identifier used to find same reports uploaded multiple times (e.g. via bug). There is no logic on backend that would process that in any way. Puppet reports do have unique UUIDs so I thought it would be good idea to require that. Now I am thinking about this, let's just drop it, there is no reason to generate this.